packages/backend/src/server/web/index.ts

```diff
@@ -173,20 +173,20 @@ router.get('/api.json', async ctx => {
 	ctx.body = genOpenapiSpec();
 });
 
-const getFeed = async (acct: string) => {
+const getFeed = async (acct: string, withAll = false, history = 5) => {
 	const { username, host } = Acct.parse(acct);
 	const user = await Users.findOneBy({
 		usernameLower: username.toLowerCase(),
 		host: host ?? IsNull(),
 		isSuspended: false,
 	});
 
-	return user && await packFeed(user);
+	return user && await packFeed(user, withAll, history);
 };
 
 // Atom
 router.get('/@:user.atom', async ctx => {
-	const feed = await getFeed(ctx.params.user);
+	const feed = await getFeed(ctx.params.user, ctx.query.history ? true : false, parseInt(ctx.query.history) ? parseInt(ctx.query.history) : 5);
```

Comment:
I think the history parameter should be sanitized. I also think there should be an upper limit to not make this exploitable in a DoS attack.

```suggestion
	let history = parseInt(ctx.query.history, 10);
	if (isNaN(history) || history <= 0 || history > 10) {
		history = 5;
	}
	const feed = await getFeed(ctx.params.user, !!history, history);
```

Maybe consider if the 2nd parameter is necessary in this case. Or maybe also do this parsing and validating in the `getFeed` function since it is reused code.

Reply:
I would say 30 is a resonable max number? some thread may getting very long